### PR TITLE
[markdown] Fix linting rule 'no-emphasis-as-heading'

### PR DESCRIPTION
**Pull request by scinos**
### Background

After linting our Markdown files following prettier format, I'd like to introduce content-based rules from `remark`. The plugin we use (`eslint-plugin-md`) recommends [Markdown Style Guide](https://cirosantilli.com/markdown-style-guide/), and it has [a preset](https://www.npmjs.com/package/remark-preset-lint-markdown-style-guide) for it.

Unfortunately this preset doesn't have the capacity of auto-fix the problems. Instead of enabling the preset and having a ton of errors, I'll enable the rules of the preset one by one and fix each error. When all the rules are enabled we can switch the hardcoded list of rules by the preset.

### Changes

Enables rule [no-emphasis-as-heading](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-emphasis-as-heading). This rule forbids to use emphasis elements (bold or italics) to simulate headers (https://cirosantilli.com/markdown-style-guide/#emphasis-vs-headers)

```
Good:

# Foo

Bad:

**Foo**
```

### Testing
Run `./node_modules/.bin/eslint --ext .md .` and check there are no errors

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ h1 {
 
 In the above example, the `body`'s font-family will be what the user selected as the base font, or `serif` if `--font-base` is not set. Likewise, the `h1`'s font-family will be what the user selected as the headings font, or `sans-serif` if `--font-headings` is not set.
 
-**Fallback values for CSS Custom Properties**
+### Fallback values for CSS Custom Properties
```
Comment (tyxla) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/README.md:48]: I'm noticing that in some cases we replace bold instances with h3, sometimes with h2, and sometimes with h4. Is there any fixed rule that we're following when we're doing that replacement?

**Review by tyxla — APPROVED** [review]
Changes are generally looking good, I'm just not 100% sure that we're always replacing `**` instances with the right heading. Any good way to confirm that?

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/README.md (hunk 1 above)
Comment (scinos) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/README.md:48]: What I did was look at other headers in the doc and figure out which level would make more sense. In some cases it was obvious what the right choice was, but in others it was a bit subjective.

Comment (tyxla) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/README.md:48]: No harm done I guess. Not-the-right header is likely better than no header at all 😉

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@scinos please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/891252). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Pull request merged**